AuthenticatorBridgeKit/SharedCryptographyService.swift

```diff
@@ -0,0 +1,143 @@
+import CryptoKit
+import Foundation
+
+// MARK: - SharedCryptographyService
+
+/// A service for handling encrypting/decrypting items to be shared between the main
+/// Bitwarden app and the Authenticator app.
+///
+public protocol SharedCryptographyService: AnyObject {
+    /// Takes an array of `AuthenticatorBridgeItemDataModel` with encrypted data and
+    /// returns the list with each member decrypted.
+    ///
+    /// - Parameter items: The encrypted array of items to be decrypted
+    /// - Returns: the array of items with their data decrypted
+    /// - Throws: AuthenticatorKeychainServiceError.keyNotFound if the Authenticator
+    ///     key is not in the shared repository.
+    ///
+    func decryptAuthenticatorItems(
+        _ items: [AuthenticatorBridgeItemDataModel]
+    ) async throws -> [AuthenticatorBridgeItemDataModel]
+
+    /// Takes an array of `AuthenticatorBridgeItemDataModel` with decrypted data and
+    /// returns the list with each member encrypted.
+    ///
+    /// - Parameter items: The decrypted array of items to be encrypted
+    /// - Returns: the array of items with their data encrypted
+    /// - Throws: AuthenticatorKeychainServiceError.keyNotFound if the Authenticator
+    ///     key is not in the shared repository.
+    ///
+    func encryptAuthenticatorItems(
+        _ items: [AuthenticatorBridgeItemDataModel]
+    ) async throws -> [AuthenticatorBridgeItemDataModel]
```

Comment:
🤔 Would it be worth keeping the same approach we have on the apps where the decrypted object name has `View` as a suffix? What do you think? cc: @KatherineInCode 
I personally would like to so we maintain the same approach everywhere but would like to hear your thoughts on this.
```swift
func decryptAuthenticatorItems(
    _ items: [AuthenticatorBridgeItemDataModel]
) async throws -> [AuthenticatorBridgeItemView]

func encryptAuthenticatorItems(
    _ items: [AuthenticatorBridgeItemView]
) async throws -> [AuthenticatorBridgeItemDataModel]
```

Reply:
I think of these as part of the syncing layer and the View portion as the responsibility of the calling apps - i.e. in the Authenticator app when we consume this, we're definitely going to be mapping these from models to views for actual display. And by not putting a View object here, we can map them directly into the View objects that already exist over there.

But I'm definitely open if you'd rather do that here. I do like the distinction of two different objects so we're not confused if this is the encrypted one or not.

Reply:
With [the CryptographyService in BWA](https://github.com/bitwarden/authenticator-ios/blob/main/AuthenticatorShared/Core/Vault/Services/CryptographyService.swift#L42) I tried to follow the convention of "Encrypting goes from `thingView` to `thing` and decrypting goes from `thing` to `thingView`" that's essentially how the SDK handles it (despite my finding the use of "view" for something that isn't a View confusing).

Having the cryptography service be `AuthenticatorBridgeItemDataModel-> AuthenticatorBridgeItemDataModel` is in my opinion going to be very confusing and lead to problems, because now you have to keep track of whether you have an _encrypted_ object or _decrypted_ object. I'd much rather have two different types, so that when you look at code you always know what you're dealing with.

Reply:
ℹ️ AFAIK here the term "View" is used as in databases where you have the plain table / data model and then a "View" which is a virtual "table" / object which is often used so it's ready to be used for higher layers without having to use the whole table/data model or having to perform additional logic which is taken care in the DB, amongst other reasons. So basically is a domain "View" not a UI view, but yes the mix is a bit confusing.

✅  I agree as well on using the suffix `View` here as is, in fact, a decrypted object which is easier to track and comprehend comparing it to the main apps code.

Reply:
Oh, yeah, "View" is definitely a term used in databases, and also has a meaning in server-side applications. It's just that "View" has a very particular sort of meaning in the iOS ecosystem (and its attempt at an MVC architecture) also, so the doubling-up was *very* confusing to me when I first got into the codebase.

It is, however, our convention. Thanks, computer science, for doubling up on terminology 😂